### PR TITLE
feat: signup redirect query param after onboarding

### PR DESCRIPTION
**Pull request by sean-brydon**
Loom: https://www.loom.com/share/6fc8126dcac1402fba236a45a64c08ff

fixes: CAL-3054 #13460 

Following the merge of this PR: https://github.com/calcom/website/pull/684 should also be merged to update pricing URL in website.

### CODE REVIEW

File: apps/web/components/getting-started/steps-views/UserProfile.tsx [hunk 1]
```diff
@@ -50,9 +50,11 @@ const UserProfile = () => {
           console.error(error);
         }
 
-        await utils.viewer.me.refetch();
-        router.push("/");
-      }
+      await utils.viewer.me.refetch();
+      const redirectUrl = localStorage.getItem("onBoardingRedirect");
+      localStorage.removeItem("onBoardingRedirect");
```
Comment (sean-brydon) [inline apps/web/components/getting-started/steps-views/UserProfile.tsx:55]: Remove from storage so this doesnt happen if the user onboards again for whatever reason.

File: apps/web/lib/signup/getServerSideProps.tsx [hunk 2]
```diff
@@ -27,8 +27,19 @@ export const getServerSideProps = async (ctx: GetServerSidePropsContext) => {
...
   const ssr = await ssrInit(ctx);
   const token = z.string().optional().parse(ctx.query.token);
+  const redirectUrlData = z
+    .string()
+    .refine((value) => value.startsWith(WEBAPP_URL), {
+      params: (value: string) => ({ value }),
+      message: "Redirect URL must start with 'cal.com'",
+    })
+    .optional()
+    .safeParse(ctx.query.redirect);
```
Comment (sean-brydon) [inline apps/web/lib/signup/getServerSideProps.tsx:37]: Ensure this URL is only to WEBAPP

File: apps/web/lib/signup/getServerSideProps.tsx [hunk 3]
```diff
@@ -27,8 +27,19 @@ export const getServerSideProps = async (ctx: GetServerSidePropsContext) => {
...
+
+  const redirectUrl = redirectUrlData.success ? redirectUrlData.data : undefined;
```
Comment (sean-brydon) [inline apps/web/lib/signup/getServerSideProps.tsx:39]: If this fails - use undefined and don't throw error

**Review by Udit-takkar — COMMENTED** [review]
@sean-brydon ~callbackUrl also does the same thing~ No it doesn't

**Comment by sean-brydon** [comment]
> @sean-brydon callbackUrl also does the same thing

o.o

**Comment by sean-brydon** [comment]
> @sean-brydon ~callbackUrl also does the same thing~ No it doesn't

As discussed in DMs - will update here for history sake:

callbackUrl happens straight after the user has signed up and this solution happens after onboarding. Me and Udit discussed in DMs that callbackUrl should probably do anyway - I will look at this

**Review by Udit-takkar — APPROVED** [review]
LGTM. Works fine

**Pull request merged**